numpy/core/src/common/simd/neon/math.h

```diff
@@ -0,0 +1,60 @@
+#ifndef NPY_SIMD
+    #error "Not a standalone header"
+#endif
+
+#ifndef _NPY_SIMD_NEON_MATH_H
+#define _NPY_SIMD_NEON_MATH_H
+
+/***************************
+ * Elementary
+ ***************************/
+// Absolute
+#define npyv_abs_f32 vabsq_f32
+#define npyv_abs_f64 vabsq_f64
+
+// Square
+NPY_FINLINE npyv_f32 npyv_square_f32(npyv_f32 a)
+{ return vmulq_f32(a, a); }
+#if NPY_SIMD_F64
+    NPY_FINLINE npyv_f64 npyv_square_f64(npyv_f64 a)
+    { return vmulq_f64(a, a); }
+#endif
+
+// Square root
+#if NPY_SIMD_F64
+    #define npyv_sqrt_f32 vsqrtq_f32
+    #define npyv_sqrt_f64 vsqrtq_f64
+#else
+    // Based on ARM doc, see https://developer.arm.com/documentation/dui0204/j/CIHDIACI
+    NPY_FINLINE npyv_f32 npyv_sqrt_f32(npyv_f32 a)
+    {
+        const npyv_f32 zero = vdupq_n_f32(0.0f);
+        // estimate to (1/√a)
+        npyv_f32 rsqrte = vrsqrteq_f32(a);
+        // fine-tune it with Newton-Raphson and vrsqrtsq_f32((3.0 + (a * -square)) / 2.0)
+        rsqrte = vmulq_f32(rsqrte, vrsqrtsq_f32(a, vmulq_f32(rsqrte, rsqrte)));
+        rsqrte = vmulq_f32(rsqrte, vrsqrtsq_f32(a, vmulq_f32(rsqrte, rsqrte)));
+        // a * (1/√a)
+        npyv_f32 sqrt = vmulq_f32(a, rsqrte);
+        // return zero if the a is zero
+        npyv_u32 bits = vbicq_u32(vreinterpretq_u32_f32(sqrt), vceqq_f32(a, zero));
```

Comment:
Floating point to integer conversions are not particularly cheap, I didn't see the zero check in OpenCV.  Can we skip this conversion?

Reply:
This new intrinsic is part of #16247.

> Floating point to integer conversions are not particularly cheap

There's no truncate conversion here,  `vreinterpretq_u32_f32()` is just perform a normal
reinterpret casting during compile time which is zero cost.

> I didn't see the zero check in OpenCV.  Can we skip this conversion?

IDK, really? No we can't since the square root of zero is "zero".

Reply:
The OpenCV simd `sqrt` is [here](https://github.com/opencv/opencv/blob/master/modules/core/include/opencv2/core/hal/intrin_neon.hpp#L887). but you are right, if we didn't check zero, the result will become a `nan`.